File: app/src/main/java/com/gameondigital/gameonapp/Login/interactor/LoginInteractorContract.java
package com.gameondigital.gameonapp.Login.interactor;

import com.gameondigital.gameonapp.Login.ui.LoginActivity;
import com.google.firebase.auth.FirebaseUser;

public interface LoginInteractorContract {
    FirebaseUser getCurrentUser();

    void setPersistenceFirebaseDatabase();

    void initFirebaseAuth();

    void login(LoginActivity loginActivity, String email, String password);
}

File: app/src/main/java/com/gameondigital/gameonapp/Login/interactor/LoginInteractorContract.kt
package com.gameondigital.gameonapp.Login.interactor

import android.app.Activity
import android.widget.EditText
import com.google.firebase.auth.FirebaseUser

interface LoginInteractorContract {
    fun initFirebaseAuth()
    fun initComponents()
    fun validateLogin(email: EditText, password: EditText)
    fun setupLogin(email: String, password: String, activity: Activity)
    fun getCurrentUser(): FirebaseUser?
}